From c170bf97c6033498a971ed52619cecc497da1748 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 22 Apr 2024 11:44:37 +0200
Subject: [PATCH 09/30] winegstreamer: Fixate caps in the pad-added callback.

Pad caps might not be fixed caps, and non-fixed caps can't be transformed to a wg_format.
---
 dlls/winegstreamer/wg_parser.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index b697dae026d..7216b640ca1 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -1209,7 +1209,7 @@ static void pad_added_cb(GstElement *element, GstPad *pad, gpointer user)
 {
     struct wg_parser_stream *stream;
     struct wg_parser *parser = user;
-    GstCaps *caps;
+    GstCaps *caps, *caps2;
 
     GST_LOG("parser %p, element %p, pad %p.", parser, element, pad);
 
@@ -1220,7 +1220,10 @@ static void pad_added_cb(GstElement *element, GstPad *pad, gpointer user)
         return;
 
     caps = gst_pad_query_caps(pad, NULL);
-    wg_format_from_caps(&stream->codec_format, caps);
+    /* uridecodebin can provide non-fixed pad caps, fixate them here */
+    caps2 = gst_caps_fixate(gst_caps_copy_nth(caps, 0));
+    wg_format_from_caps(&stream->codec_format, caps2);
+    gst_caps_unref(caps2);
     gst_caps_unref(caps);
 
     /* For compressed stream, create an extra decodebin to decode it. */
-- 
2.44.0

